### ISSUE TITLE
image resize before upload does not work

### ISSUE THREAD

**Issue by quape** [orthes/medium-editor-insert-plugin#288]
Hi there,

as this plugin uses blueimp fileupload for uplading images I though I could just add this to `fileUploadOptions`:

    disableImageResize: false,
    imageMaxWidth: 400,
    imageMaxHeight: 400

(Of course I added the depending js libs to my app)
Unfortunately that does not work at all. The file is uploaded but not resized before. Are there maybe some interfering things with the preview function of this plugin and the blueimp process() ?

Thanks

**Comment by orthes** [comment]
No, client size resizing is not supported right now. The only option is to resize on server and return to the plugin resized imaged.

**Comment by quape** [comment]
Well, you use blueimp fileupload which supports resizing. So it should only be some little piece of code to make it work. If you give me a hint why it doesn't work out of the box I can do it.

**Comment by orthes** [comment]
I don't know how blueimp's resizing works. In our plugin we use [`Images.showImage(img, data)`](https://github.com/orthes/medium-editor-insert-plugin/blob/2.2.2/src/js/images.js#L358) function to add an image to content. 

This function is called twice:

1. If you have `preview` enabled, it is called from [`Images.uploadAdd()`](https://github.com/orthes/medium-editor-insert-plugin/blob/2.2.2/src/js/images.js#L274) function with `img` parameter set to preview image data BLOB.
2. When upload is done, the function is called one more time from [`Images.uploadDone()`](https://github.com/orthes/medium-editor-insert-plugin/blob/2.2.2/src/js/images.js#L345) function where it is called with `img` parameter set to URL to uploaded image and if the `preview` was enabled, the preview image is just replaced with the real image URL.

So my guess would be to take a look at [`Images.uploadDone()`](https://github.com/orthes/medium-editor-insert-plugin/blob/2.2.2/src/js/images.js#L345) and change `data.result.files[0].url` to resized image provided by jQuery File Upload plugin.

**Comment by amboleos** [comment]
I have found a solution as below with changing "medium-editor-insert-plugin.js"
```

fileUploadOptions = {
                dataType: 'json',
                add: function (e, data) {
                    $.proxy(that, 'uploadAdd', e, data)();
                    $.blueimp.fileupload.prototype.options.add.call(this, e, data); // Add this line
                },
                done: function (e, data) {
                    $.proxy(that, 'uploadDone', e, data)();
                },
                disableImageResize: false,  // Add this line
                imageMaxWidth: 960,  // Add this line
            };
```
